File: frontend/src/app/models/user-update-request.ts

export class UserUpdateRequest {

  public id: number;
  public desiredCalories: number
  public email: string

  constructor( o: any){
    this.id = o.id;
    this.desiredCalories = o.desiredCalories;
    this.email = o.email;
  }

  public toRequestJson() :any{
    return {
      desiredCalories: this.desiredCalories,
      email: this.email
    }
  }
}

File: backend/src/main/java/io/abnd/model/UserUpdateRequest.java
package io.abnd.model;

import java.util.Optional;
import java.util.Set;

public class UserUpdateRequest {
  private Optional<Integer> desiredCalories;
  private String email;

  public Optional<Integer> getDesiredCalories() {
    return desiredCalories;
  }

  public void setDesiredCalories(final Optional<Integer> desiredCalories) {
    this.desiredCalories = desiredCalories;
  }

  public String getEmail() {
    return email;
  }

  public void setEmail(final String email) {
    this.email = email;
  }
}
